--- /dev/null	2017-11-05 09:30:41.941867500 +0100
+++ ./scripts/config/Makefile	2017-11-05 09:31:19.424187665 +0100
@@ -39,7 +39,7 @@
 
 conf: $(conf-objs)
 mconf: $(mconf-objs) $(lxdialog-objs)
-	$(CC) -o $@ $^ $(call check_lxdialog,ldflags $(CC))
+	$(CC) -o $@ $^ $(LIBS) -ltinfo $(call check_lxdialog,ldflags $(CC))
 qconf: $(qconf-cxxobjs) $(qconf-objs)
 ifneq ($(DISTRO-PKG-CONFIG),)
 	$(CXX) $(HOSTLOADLIBES_qconf) -o $@ $^
