--- a/target/linux/ar71xx/Makefile
+++ b/target/linux/ar71xx/Makefile
@@ -10,7 +10,7 @@
 BOARD:=ar71xx
 BOARDNAME:=Atheros AR7xxx/AR9xxx
 FEATURES:=usbgadget
-CPU_TYPE:=24kc
+CPU_TYPE:=74kc
 SUBTARGETS:=generic tiny nand mikrotik
 
 KERNEL_PATCHVER:=4.9
@@ -19,6 +19,6 @@
 
 DEFAULT_PACKAGES += \
 	kmod-gpio-button-hotplug swconfig \
-	kmod-ath9k wpad-mini uboot-envtools
+	kmod-ath9k wpad
 
 $(eval $(call BuildTarget))
