diff --git a/package/libs/openssl/Config.in b/package/libs/openssl/Config.in
index dbcd11a..07666ca 100644
--- a/package/libs/openssl/Config.in
+++ b/package/libs/openssl/Config.in
@@ -17,7 +17,7 @@ config OPENSSL_WITH_SSL3
 
 config OPENSSL_WITH_DEPRECATED
 	bool
-	default y
+	default y
 	prompt "Include deprecated APIs"
 
 config OPENSSL_WITH_DTLS
@@ -50,6 +50,12 @@ config OPENSSL_ENGINE_DIGEST
 	depends on OPENSSL_ENGINE_CRYPTO
 	prompt "Digests acceleration support"
 
+config OPENSSL_THREADS
+	bool
+	prompt "Threading support"
+	default y
+	select PACKAGE_libpthread
+
 config OPENSSL_HARDWARE_SUPPORT
 	bool
 	default n
diff --git a/package/libs/openssl/Makefile b/package/libs/openssl/Makefile
index 9b64302..1bc9e85 100644
--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -34,6 +34,7 @@ PKG_CONFIG_DEPENDS:= \
 	CONFIG_OPENSSL_WITH_EC2M \
 	CONFIG_OPENSSL_WITH_SSL3 \
 	CONFIG_OPENSSL_HARDWARE_SUPPORT \
+	CONFIG_OPENSSL_THREADS \
 	CONFIG_OPENSSL_WITH_DEPRECATED \
 	CONFIG_OPENSSL_WITH_DTLS \
 	CONFIG_OPENSSL_WITH_COMPRESSION \
@@ -127,6 +128,12 @@ ifndef CONFIG_OPENSSL_HARDWARE_SUPPORT
   OPENSSL_OPTIONS += no-hw
 endif
 
+ifdef CONFIG_OPENSSL_THREADS
+  OPENSSL_OPTIONS += threads
+else
+  OPENSSL_OPTIONS += no-threads
+endif
+
 ifndef CONFIG_OPENSSL_WITH_DEPRECATED
   OPENSSL_OPTIONS += no-deprecated
 endif
