--- ./package/qca/fast-classifier/Makefile	2017-06-22 20:27:21.576534699 +0800
+++ ./package/qca/fast-classifier/Makefile	2017-06-22 19:08:41.359655708 +0800
@@ -6,7 +6,7 @@ PKG_SOURCE_PROTO:=git
 PKG_BRANCH:=master
 PKG_RELEASE:=1
 
-LOCAL_SRC:=$(TOPDIR)/qca/src/shortcut-fe
+LOCAL_SRC:=$(TOPDIR)/package/qca/shortcut-fe/src/
 include $(INCLUDE_DIR)/local-development.mk
 ifeq ($(DUMP)$(PKG_VERSION),)
   PKG_REV:=$(shell git ls-remote $(PKG_SOURCE_URL) $(PKG_BRANCH) | cut -b -7)
@@ -23,7 +23,7 @@ define KernelPackage/$(PKG_NAME)/Default
   SECTION:=kernel
   CATEGORY:=Kernel modules
   SUBMENU:=Network Support
-  DEPENDS:=+kmod-ipt-conntrack +kmod-shortcut-fe +kmod-ipv6
+  DEPENDS:=+kmod-ipt-conntrack +kmod-shortcut-fe
   TITLE:=Kernel driver for FAST Classifier
   FILES:=$(PKG_BUILD_DIR)/fast-classifier/fast-classifier.ko
   KCONFIG:=CONFIG_NF_CONNTRACK_CHAIN_EVENTS=y
@@ -70,7 +70,7 @@ define Build/Compile/kmod
 		CROSS_COMPILE="$(TARGET_CROSS)" \
 		ARCH="$(LINUX_KARCH)" \
 		SUBDIRS="$(PKG_BUILD_DIR)/fast-classifier" \
-		EXTRA_CFLAGS="$(EXTRA_CFLAGS)" \
+		EXTRA_CFLAGS="$(EXTRA_CFLAGS) -Wno-implicit-function-declaration" \
 		modules
 endef
 
--- ./package/qca/simulated-driver/Makefile	2017-06-22 20:28:20.392708527 +0800
+++ ./package/qca/simulated-driver/Makefile	2017-06-22 20:28:11.634682643 +0800
@@ -20,7 +20,7 @@ PKG_SOURCE_PROTO:=git
 PKG_BRANCH:=master
 PKG_RELEASE:=1
 
-LOCAL_SRC:=$(TOPDIR)/qca/src/shortcut-fe
+LOCAL_SRC:=$(TOPDIR)/package/qca/shortcut-fe/src/
 include $(INCLUDE_DIR)/local-development.mk
 ifeq ($(DUMP)$(PKG_VERSION),)
   PKG_REV:=$(shell git ls-remote $(PKG_SOURCE_URL) $(PKG_BRANCH) | cut -b -7)
--- ./package/qca/shortcut-fe/Makefile	2017-06-22 20:54:20.550640047 +0800
+++ ./package/qca/shortcut-fe/Makefile	2017-06-22 20:54:12.611616294 +0800
@@ -57,7 +57,7 @@ define KernelPackage/shortcut-fe-cm
   SECTION:=kernel
   CATEGORY:=Kernel modules
   SUBMENU:=Network Support
-  DEPENDS:=+kmod-ipt-conntrack +kmod-shortcut-fe +kmod-ipv6
+  DEPENDS:=+kmod-ipt-conntrack +kmod-shortcut-fe
   TITLE:=Kernel driver for SFE
   FILES:=$(PKG_BUILD_DIR)/shortcut-fe/shortcut-fe-cm.ko
   KCONFIG:=CONFIG_NF_CONNTRACK_CHAIN_EVENTS=y
