--- a/package/qca/qca-ssdk/src/make/config.mk	2017-03-26 14:19:18.000000000 +0800
+++ b/package/qca/qca-ssdk/src/make/config.mk	2017-03-26 14:20:41.955052618 +0800
@@ -59,7 +59,7 @@ else
      ifneq (TRUE, $(FAL))
          $(error FAL must be TRUE when CHIP_TYPE is defined as ALL_CHIP!)
      endif
-     SUPPORT_CHIP = ISIS ISISC SHIVA DESS
+     SUPPORT_CHIP = ISIS ISISC
   endif
 
   ifndef SUPPORT_CHIP
